azurerm/internal/services/policy/resource_arm_policy_remediation.go

```diff
@@ -0,0 +1,248 @@
+	client := meta.(*clients.Client).Policy.RemediationsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.PolicyRemediationID(d.Id())
+	if err != nil {
+		return fmt.Errorf("unable to read Policy Remediation: %+v", err)
+	}
+
+	resp, err := RemediationGetAtScope(ctx, client, id.Name, id.PolicyScopeId)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Policy Remediation %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("unable to read Policy Remediation %q (Scope %q): %+v", id.Name, id.ScopeId(), err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("scope", id.ScopeId())
+
+	if props := resp.RemediationProperties; props != nil {
+		if err := d.Set("location_filters", flattenArmRemediationLocationFilters(props.Filters)); err != nil {
+			return fmt.Errorf("unable to set `location_filters`: %+v", err)
+		}
+
+		d.Set("policy_assignment_id", props.PolicyAssignmentID)
+		d.Set("policy_definition_reference_id", props.PolicyDefinitionReferenceID)
+	}
+
+	return nil
+}
+
+func resourceArmPolicyRemediationDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Policy.RemediationsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.PolicyRemediationID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	switch scope := id.PolicyScopeId.(type) {
+	case parse.ScopeAtSubscription:
+		_, err = client.DeleteAtSubscription(ctx, scope.SubscriptionId, id.Name)
+	case parse.ScopeAtResourceGroup:
+		_, err = client.DeleteAtResourceGroup(ctx, scope.SubscriptionId, scope.ResourceGroup, id.Name)
+	case parse.ScopeAtResource:
+		_, err = client.DeleteAtResource(ctx, scope.ScopeId(), id.Name)
+	case parse.ScopeAtManagementGroup:
+		_, err = client.DeleteAtManagementGroup(ctx, scope.ManagementGroupId, id.Name)
+	default:
+		return fmt.Errorf("unable to delete Policy Remediation %q: invalid scope type", id.Name)
+	}
+	if err != nil {
+		return fmt.Errorf("unable to delete Policy Remediation %q (Scope %q): %+v", id.Name, id.ScopeId(), err)
+	}
+
+	return nil
+}
+
+func expandArmRemediationLocationFilters(input []interface{}) *policyinsights.RemediationFilters {
+	if len(input) == 0 {
+		return nil
+	}
+
+	result := policyinsights.RemediationFilters{
+		Locations: utils.ExpandStringSlice(input),
+	}
+
+	return &result
+}
+
+func flattenArmRemediationLocationFilters(input *policyinsights.RemediationFilters) []interface{} {
+	if input == nil {
+		return make([]interface{}, 0)
+	}
+
+	return utils.FlattenStringSlice(input.Locations)
+}
+
+// RemediationGetAtScope is a wrapper of the 4 Get functions on RemediationsClient, combining them into one to simplify code.
+func RemediationGetAtScope(ctx context.Context, client *policyinsights.RemediationsClient, name string, scopeId parse.PolicyScopeId) (policyinsights.Remediation, error) {
```

Comment:
Should this be private to this package?
```suggestion
func remediationGetAtScope(ctx context.Context, client *policyinsights.RemediationsClient, name string, scopeId parse.PolicyScopeId) (policyinsights.Remediation, error) {
```

Reply:
Emmm I am afraid that we cannot do this... because this is also used in the acc tests, the checkDestroy or checkExist functions are using this to query whether the remediation is successfully provisioned.

Reply:
👍